File: DS-Algo-with-Java/Lecture-9/Complex.java
import java.util.Scanner;

class Calculation{
    int real;
    int imag;
    public Calculation(int r, int i){
        real = r;
        imag = i;
    }
    public static Calculation add(Calculation a, Calculation b){
        return new Calculation((a.real+b.real),(a.imag+b.imag));
    }
    public static Calculation diff(Calculation a, Calculation b){
        return new Calculation((a.real-b.real),(a.imag-b.imag));
    }
    public static Calculation product(Calculation a, Calculation b){
        return new Calculation(((a.real*b.real)-(a.imag*b.imag)),((a.real*b.imag)+(a.imag*b.real)));
    }
    public void printComplex(){
        if(real == 0 && imag!=0){
            System.out.println(imag+"i");
        }
        else if(imag == 0 && real!=0){
            System.out.println(real);
        }
        else{
            System.out.println(real+"+"+imag+"i");
        }
    }
}

class Complex{
    public static void main(String[] args){
        try (Scanner sc = new Scanner(System.in)) {
            System.out.print("Enter real part of complex number 1: ");
            int real1 = sc.nextInt();
            System.out.print("Enter complex part of complex number 1 without iota sign: ");
            int comp1 = sc.nextInt();
            System.out.print("Enter real part of complex number 2: ");
            int real2 = sc.nextInt();
            System.out.print("Enter complex part of complex number 2 without iota sign: ");
            int comp2 = sc.nextInt();
            Calculation c = new Calculation(real1, comp1);
            Calculation d = new Calculation(real2, comp2);
            Calculation e = Calculation.add(c,d);
            Calculation f = Calculation.diff(c,d);
            Calculation g = Calculation.product(c,d);
            e.printComplex();
            f.printComplex();
            g.printComplex();
        }
    }
}

File: DS-Algo-with-Python/Lecture-9/Complex.py
class Calculation:
    real = None
    imag = None

    def __init__(self, r, i):
        self.real = r
        self.imag = i
    
    def add(a, b):
        return Calculation((a.real + b.real), (a.imag + b.imag))
    
    def diff(a, b):
        return Calculation((a.real - b.real), (a.imag - b.imag))
    
    def product(a, b):
        return Calculation((a.real * b.real) - (a.imag * b.imag), (a.real * b.imag) + (a.imag * b.real))
    
    def printComplex(self):
        if self.real == 0 and self.imag != 0:
            print(self.imag + "i")
        elif self.imag == 0 and self.real != 0:
            print(self.real)
        else:
            print(f"{self.real} + {self.imag}i")

real1 = int(input("Enter real part of complex number 1: "))
comp1 = int(input("Enter complex part of complex number 1 without iota sign: "))
real2 = int(input("Enter real part of complex number 2: "))
comp2 = int(input("Enter complex part of complex number 2 without iota sign: "))

c = Calculation(real1, comp1)
d = Calculation(real2, comp2)
e = Calculation.add(c, d)
f = Calculation.diff(c, d)
g = Calculation.product(c, d)

e.printComplex()
f.printComplex()
g.printComplex()
